packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -370,7 +374,7 @@ export class Distribution extends Resource implements IDistribution {
         this.addOriginGroup(originGroupId, originBindConfig.failoverConfig.statusCodes, originId, failoverOriginId);
         return originGroupId;
       }
-      return originId;
+      return originBindConfig.originProperty?.id ?? originId;
```

Comment:
Is this necessary? `originBindConfig.originProperty.id === originId` under our implementation of `OriginBase.bind()`.

I suppose someone could write their own `CustomOrigin extends IOrigin` and implement `bind` where the given `originId` does not become the `originBindConfig.originProperty.id`, but is that what we are defending against here?

If so... fine. But these will 99% of the time be the same value.

Reply:
They will be different if the user provides their own ID.

For example.

```ts
const origin = new S3Origin(bucket, { originId: 'MyCustomOrigin' });
```

When we then call bind on line `358` we pass in the generated originId (from line `357`. The change I made in `bind` is to take the originId passed in `S3Origin` first. So originBindConfig.originProperty.id will be `MyCustomOrigin`.